src/game/boe.dlgutil.cpp

```diff
@@ -881,8 +888,10 @@ void handle_talk_node(int which_talk_entry) {
 			return;
 			
 		case eTalkNode::SHOP:
-			start_shop_mode(b,a,save_talk_str1);
-			can_save_talk = false;
+			if(!start_shop_mode(b,a,save_talk_str1,true))
+				save_talk_str1 = "There is nothing to buy from " + save_talk_str1 + ".";
```

Comment:
It looks like shops don't use the second text string from the dialogue node. We could use that here, maybe? Although it may be better to use it as a string shown when shopping concludes… I'm not sure.

(Whether we use it here or not, providing this is a reasonable default.)

Reply:
I think it'd be nicer to customize what a shopkeeper says when they can't offer you anything, than when you finish shopping. When there's nothing to buy they could apologize, say goods are in a shortage, other interesting and specific stuff. Healers could say "You look fine to me!" and stuff like that.

Reply:
I've made the change.